toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1753,6 +1898,56 @@ func createRawArtifact(workDirPath, devPath, name string) (err error) {
 	return shell.ExecuteLive(squashErrors, "dd", ddArgs...)
 }
 
+func createRDiffArtifact(workDirPath, devPath, rDiffBaseImage, name string) (err error) {
+	const (
+		signatureFileName = "./signature"
+		squashErrors      = true
+	)
+
+	fullPath := filepath.Join(workDirPath, name)
+
+	rdiffArgs := []string{
+		"signature",
+		rDiffBaseImage,
+		signatureFileName,
+	}
+
+	err = shell.ExecuteLive(squashErrors, "rdiff", rdiffArgs...)
+	if err != nil {
+		return
+	}
+	defer os.Remove(signatureFileName)
+
+	rdiffArgs = []string{
+		"delta",
+		signatureFileName,
+		devPath,
+		fullPath,
+	}
+
+	return shell.ExecuteLive(squashErrors, "rdiff", rdiffArgs...)
+}
+
+// randomString generates a random string of the length specified
+// using the provided legalCharacters.  crypto.rand is more secure
+// than math.rand and does not need to be seeded.
+func randomString(length int, legalCharacters string) (output string, err error) {
```

Comment:
The method and its usage can be replaced with the recently added `randomization. RandomString(...)`:

https://github.com/microsoft/CBL-Mariner/blob/124daab644bcb9ed7ab65a94d44a5694b5ce569d/toolkit/tools/internal/randomization/randomization.go#L16

Reply:
Fixed.

---
In reply to: [573082120](https://github.com/microsoft/CBL-Mariner/pull/611#discussion_r573082120) [](ancestors = 573082120)